pytorch/imagerecognition/openmpi-cifar10-resnet20-all-reduce-scaling/main.py

```diff
@@ -169,19 +169,30 @@ def train_loop(run_id, dataset_dir, ckpt_run_dir, output_dir,
             json.dump(val_stats, f)
 
 
-def main(run_id, dataset_dir, ckpt_run_dir, output_dir, validation_only=False,
+def main(run_id, dataset_dir, ckpt_run_dir, output_dir, rank, world_size, backend, hosts, validation_only=False,
          gpu=False, light_target=False):
     r"""Main logic."""
-
-    with initialize_backends(
-            comm_backend='mpi',
-            logging_level='INFO',
-            logging_file=os.path.join(output_dir, 'mlbench.log'),
-            use_cuda=gpu,
-            seed=42,
-            cudnn_deterministic=False,
-            ckpt_run_dir=ckpt_run_dir,
-            delete_existing_ckpts=not validation_only):
+    if backend == "mpi":
+        with initialize_backends(
```

Comment:
Ideally, modify this function so that we do not have `if else` in the benchmark code (try and reduce the code to the strict necessary)

Reply:
Yes, `initialize_backends` already has the `comm_backend` parameter which is intended for this.

Reply:
This is related to the previous comment. If we want to continue using the `initialize_backends` method as is,  we need to expose the same environment variables that are available for MPI to Gloo.